File: day1/advent1.py
#!/usr/bin/python


with open('input.txt', 'r') as f:
    seq = f.readline().rstrip()
length = len(seq)

i = 0
sum = 0
while i <= length - 1:
    j = (i + 1) % length
    if seq[i] is seq[j]:
        sum += int(seq[i])
    i += 1
print sum

i = 0
sum = 0
half = length / 2
while i <= length - 1:
    j = (i + half) % length
    if seq[i] is seq[j]:
        sum += int(seq[i])
    i += 1
print sum

File: day1/advent1.rb
#!/usr/local/bin/ruby

seq = File.read("input.txt").rstrip()
length = seq.length

sum = 0
i = 0
while i <= length - 1
	j = (i + 1) % length
	if seq[i] == seq[j]
		sum += Integer(seq[i])
	end
	i += 1
end
puts sum

sum = 0
i = 0
half = length / 2
while i <= length - 1
	j = (i + half) % length
	if seq[i] == seq[j]
		sum += Integer(seq[i])
	end
	i += 1
end
puts sum
